hibernate-core/src/main/java/org/hibernate/tool/schema/internal/ColumnDefinitions.java

```diff
@@ -26,7 +26,11 @@
 
 class ColumnDefinitions {
 
-	static boolean hasMatchingType(Column column, ColumnInformation columnInformation, Metadata metadata, Dialect dialect) {
+	static boolean hasMatchingType(Column column, ColumnInformation columnInformation, Metadata metadata, Dialect dialect, boolean forUpdate) {
+		if ( forUpdate && column.getColumnDefinition() != null ) {
+			// validation shouldn't use columnDefinition
+			return extractTypeName( column.getColumnDefinition() ).equals( normalize( columnInformation.getTypeName() ) );
```

Comment:
I don't see how this is going to work well at all on databases like Postgres or Spanner which have native type names like `timestamptz`, `int8`, etc. 

To make this approach work, you would need to do dialect-specific normalization. The existing `normalize()` function only knows about variations within ANSI SQL.

Reply:
> To make this approach work, you would need to do dialect-specific normalization. The existing `normalize()` function only knows about variations within ANSI SQL.

You are right, but it's an onerous task but doesn't gain too much value, I'd rather not to invest that.

Reply:
Ok but that investment that you're not prepared to make is something someone else is effort that other people are going to have to put into maintaining this. I happen to agree with you that the value of this does not justify the maintenance cost.